src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -1516,11 +1818,80 @@ private function step_in_body() {
 				return true;
 
 			/*
-			 * > A start tag whose tag name is one of: "param", "source", "track"
+			 * > A start tag whose tag name is "image"
 			 */
-			case '+PARAM':
-			case '+SOURCE':
-			case '+TRACK':
+			case '+IMAGE':
+				/*
+				 * > Parse error. Change the token's tag name to "img" and reprocess it. (Don't ask.)
+				 *
+				 * Note that this is handled elsewhere, so it should not be possible to reach this code.
+				 */
+				$this->last_error = self::ERROR_UNSUPPORTED;
+				throw new WP_HTML_Unsupported_Exception( "Cannot process an IMAGE tag. (Don't ask.)" );
+
+			/*
+			 * > A start tag whose tag name is "textarea"
+			 */
+			case '+TEXTAREA':
+				$this->insert_html_element( $this->state->current_token );
+
+				/*
+				 * > If the next token is a U+000A LINE FEED (LF) character token, then ignore
+				 * > that token and move on to the next one. (Newlines at the start of
+				 * > textarea elements are ignored as an authoring convenience.)
+				 *
+				 * This is handled in `get_modifiable_text()`.
+				 */
+
+				$this->state->frameset_ok = false;
+
+				/*
+				 * > Switch the insertion mode to "text".
+				 *
+				 * As a self-contained node, this behavior is handled in the Tag Processor.
+				 */
+				return true;
+
+			/*
+			 * > A start tag whose tag name is "xmp"
+			 */
+			case '+XMP':
+				if ( $this->state->stack_of_open_elements->has_p_in_button_scope() ) {
+					$this->close_a_p_element();
+				}
+
+				$this->reconstruct_active_formatting_elements();
+				$this->state->frameset_ok = false;
+
+				/*
+				 * > Follow the generic raw text element parsing algorithm.
+				 *
+				 * As a self-contained node, this behavior is handled in the Tag Processor.
+				 */
+				$this->insert_html_element( $this->state->current_token );
+				return true;
+
+			/*
+			 * A start tag whose tag name is "iframe"
+			 */
+			case '+IFRAME':
+				$this->state->frameset_ok = false;
+
+				/*
+				 * > Follow the generic raw text element parsing algorithm.
+				 *
+				 * As a self-contained node, this behavior is handled in the Tag Processor.
+				 */
+				$this->insert_html_element( $this->state->current_token );
+				return true;
+
+			/*
+			 * > A start tag whose tag name is "noembed"
+			 * > A start tag whose tag name is "noscript", if the scripting flag is enabled
+			 *
+			 * The scripting flag is never enabled in this parser.
+			 */
+			case '+NOEMBED':
```

Comment:
I'm worried about not handling `+NOSCRIPT` here. If we don't handle it, I think it ends up being handled as "any other start tag" and it does the wrong thing 🙃 

![Screenshot 2024-07-05 at 18 54 42](https://github.com/WordPress/wordpress-develop/assets/841763/0de42c41-af64-4fe6-9cdc-397f12e6bbd2)

Reply:
I'm glad you asked, but this is correct the way it is. that rule doesn't exist _in this parser_ because this parser does not enable the scripting flag.

the reason the rule is conditional is because _when the scripting flag is enabled_ the NOSCRIPT tag enters RAWTEXT, skipping any nodes that exist inside it. this makes sense, because in a browser where scripting is enabled, you don't want any parsing to occur within that element.

but we don't run JavaScript, and so if we ignore the NOSCRIPT contents then we will prevent people from modifying the fallback content. if you disable JavaScript and re-examine the DOM tree you should find that it matches the HTML Processor's output.

Reply:
With JavaScript enabled
<img width="257" alt="Screenshot 2024-07-05 at 10 48 31 AM" src="https://github.com/WordPress/wordpress-develop/assets/5431237/4b596b09-efc4-48dd-b0c6-e1f9e7b7a42a">

With JavaScript disabled
<img width="244" alt="Screenshot 2024-07-05 at 10 48 49 AM" src="https://github.com/WordPress/wordpress-develop/assets/5431237/b6bb28e9-9863-461c-ba68-8f60db45385e">

Reply:
You're right, this makes sense as-is 👍